common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookModel.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.questbook;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.Model;
+import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.model.container.ContainerContent;
+import com.wynntils.wynn.model.container.ScriptedContainerQuery;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.ArrayList;
+import java.util.List;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+public class QuestBookModel extends Model {
+    private static final int NEXT_PAGE_SLOT = 8;
+
+    private static List<QuestInfo> quests = List.of();
+    private static List<QuestInfo> newQuests;
+
+    public static List<QuestInfo> getQuests() {
+        return quests;
+    }
+
+    public static void queryQuestBook() {
+        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Quest Book Query")
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .processContainer(c -> processQuestBookPage(c, 1))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(2))
+                .matchTitle(getQuestBookTitle(2))
+                .processContainer(c -> processQuestBookPage(c, 2))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(3))
+                .matchTitle(getQuestBookTitle(3))
+                .processContainer(c -> processQuestBookPage(c, 3))
+                .clickOnSlotMatching(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(4))
+                .matchTitle(getQuestBookTitle(4))
```

Comment:
Is there no way to beat this hard coded values?

Reply:
What do you mean? I have explicitly tried to make the matching as exact as possible, to minimize the risk that we're randomly clicking around in some incorrect container. There was some war stories about folks who've lost mythics due to the Legacy fake inventory being buggy, so I want this to be "better safe than sorry". That means I try to match exactly what I expect, and if it does not match, I will fail.

Reply:
I mean the fact that we've specified each quest book page's name

Reply:
That is very much by purpose. I could have skipped the title matching completely, or made it just a more fuzzy regexp match, but I really wanted to make sure we go get to exactly the page we want.

This is for instance how I discovered that some quest pages where sent twice -- I sent the "click on 'go to page 2'" packet, got an update, but it was still for page 1.